ddd-koans/Entity/Lifecycle/task.html

```diff
@@ -23,5 +23,17 @@ <h2>Lifecycle</h2>
     To sum up, a bar can not yet exist, be in an incomplete state or be in a complete state.
     The lifecycle must be modelled in the code then.
 </p>
-<h2>Examples</h2>
+<h2>Task</h2>
+<p>
+    To begin with the lifecycle, please implement <code>Bar#state()</code> method which returns
+    Bar.State.Complete for a complete bar, Bar.State.Incomplete for an incomplete bar and throws
+    BarWithTooManyNotesException when there are too many notes in the bar.
```

Comment:
depending on how our model is going evaluate, we may have a situation that:

1. we have a bar in 4/4
2. we have a function to change Bar's time signature -> we change time signature to 3/4
3. do we drop redundant notes from the end/beginning or maybe allow to have state with redundant notes? 🤔

Reply:
1. What is wrong in bar 4/4, from a tempo perspective this is not the same as 1/1.
2&3. I assumed we need to first drop notes and then change the time signature, but we can add OverfulBar and allow such behavior.